gradio/components/chatbot.py

```diff
@@ -42,14 +42,14 @@ class MetadataDict(TypedDict):
         title: The title of the "thought" message. Required if the message is to be displayed as a thought.
         id: The ID of the message. Only used for nested thoughts. Nested thoughts can be nested by setting the parent_id to the id of the parent thought.
         parent_id: The ID of the parent message. Only used for nested thoughts.
-        duration: The duration of the message. Appears next to the title in the thought bubble in a subdued font.
+        duration: The duration of the message. Appears next to the thought title in a subdued font. Can be a float (which is displayed as a number of seconds) or an arbitrary string (which is displayed as is).
         status: The status of the message. If "pending", the status is displayed as a spinner icon.
     """
 
     title: NotRequired[str]
     id: NotRequired[int | str]
     parent_id: NotRequired[int | str]
-    duration: NotRequired[float]
+    duration: NotRequired[float | str]
```

Comment:
On second thought, what if we keep duration as is but add a `log` parameter which has this float/string behavior. Seeing something like `duration="256 Output Tokens"` would be confusing in code and the name `log` makes it more apparent that strings can be passed in.